### DISCUSSION TITLE
Self-hosted instance sending out emails

### DISCUSSION THREAD

**Discussion by kabatp** [ente-io/ente#4486] [Q&A] [answered]
Hello,
In search for the verification code, I found the following in the documentation: "The self-hosted Ente by default does not send out emails" - [https://help.ente.io/self-hosting/faq/otp](https://help.ente.io/self-hosting/faq/otp)

Does this mean there is a way to set up SMTP for Ente's self-hosted server? If so, there is no other mention of it in the documentation. It would be useful for legacy, verification codes, password resets, etc. if this option would exist in the self-hosted version.

**Comment by mnvr** [comment, marked as answer]
> Does this mean there is a way to set up SMTP for Ente's self-hosted server? 

Yes. "by default" it doesn't send out emails, but sure, you if you configure the SMTP settings it will send emails. You can configure it using the "smtp" block in the configuration:

https://github.com/ente-io/ente/blob/main/server/configurations/local.yaml#L213-L230

**Reply by kabatp** [reply]
Thank you! The setup was easy and flawless!

**Reply by wrobelda** [reply]
I just set up a self-hosted instance using your upstream docker-compose and seeing this after adding a user.:

<img width="442" alt="Screenshot 2025-03-27 at 7 48 12 PM" src="https://github.com/user-attachments/assets/055a5a2d-e17e-4ef7-ab30-2a00ff128917" />

The web container log file claims otherwise, though:

`INFO[0516]email.go:130 sendViaTransmail Skipping sending email to cromo@klej.net: Verification code: 123456`

EDIT: this is actually mentioned in the docs: https://github.com/ente-io/ente/blob/main/server/docs/quickstart.md
